src/content/docs/en/guides/server-islands.mdx

```diff
@@ -33,6 +33,15 @@ const avatarURL = await getUserAvatar(userSession);
 ---
 <img alt="User avatar" src={avatarURL} />
 ```
+:::caution[Server Island Prop Limitations]
+You can pass a function as a prop to a server island component, but this only works during server rendering. If you try to use the function in a hydrated component (for example, as an event handler), an error will occur.
+
+This is because functions cannot be _serialized_ (transferred from the server to the client) by Astro. Objects with circular references are also not serializable.
```

Comment:
I think this should have the same wording regarding serialization as the client islands. Even more so here to not say that the limitation is regarding sharing information with the client because that is an implementation detail. It could send the information between instances of the server (which might remove some other limitations we currently have) and it would have the same problem.